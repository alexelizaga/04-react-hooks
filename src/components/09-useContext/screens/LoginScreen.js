import React, { useContext } from 'react';
import { UserContext } from '../context/UserContext';

export const LoginScreen = () => {
  const {user, setUser} = useContext(UserContext);

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button
        data-testid='LoginScreenBtn'
        className='btn btn-primary'
        onClick={ () => setUser({
          id: 123,
          name: 'Alex'
        })}
      >
        Login
      </button>
    </div>
  );
};
