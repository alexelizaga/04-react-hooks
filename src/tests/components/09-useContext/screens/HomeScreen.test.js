import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';

import { HomeScreen } from '../../../../components/09-useContext/screens/HomeScreen';
import { UserContext } from '../../../../components/09-useContext/context/UserContext';

describe('Test <HomeScreen />', () => {

  const user = {
    name: 'Alex',
    email: 'alex@test.com'
  }

  test('should render correctly', () => {
    const tree = renderer.create(
      <UserContext.Provider value={{
        user
      }}>
        <HomeScreen />
      </UserContext.Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

})