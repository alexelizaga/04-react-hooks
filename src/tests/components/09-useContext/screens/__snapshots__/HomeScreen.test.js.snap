// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <HomeScreen /> should render correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
    "name": "Alex",
    "email": "alex@test.com"
}
  </pre>
</div>
`;
